ouroboros-consensus-protocol/src/Ouroboros/Consensus/Protocol/Praos/Header.hs

```diff
@@ -123,6 +131,22 @@ pattern Header {headerBody, headerSig} <-
     Header body sig =
       HeaderConstr $ memoBytes (encodeHeaderRaw $ HeaderRaw body sig)
 
+{-# COMPLETE Header #-}
+
+-- | Compute the size of the header
+headerSize ::
+  (CC.Crypto crypto) =>
+  Header crypto ->
+  Int
+headerSize = BS.length . serialize'
```

Comment:
In theory, it shouldn't matter here, since I'm using the `MemoBytes` construction here (wasn't used upstream). The `ToCBOR` instance just returns the already serialised bytes. But this does entail a conversion from `ShortByteString` to `ByteString`, which is I guess expensive! I'll fix it.

Reply:
Hmm, maybe this instance is using `MemoBytes` whereas the one Ryan found wasn't 🤔